analyzer/codechecker_analyzer/analyzer.py

```diff
@@ -79,7 +79,8 @@ def __get_analyzer_version(context, analyzer_config_map):
         version = [analyzer_bin, u' --version']
         try:
             output = subprocess.check_output(shlex.split(' '.join(version)),
-                                             env=check_env)
+                                             env=check_env,
+                                             universal_newlines=True)
```

Comment:
The name of this argument is not too intuitive. The point is, these kind of subprocess calls will return `str`s in Python2 and 'byte`s in Python3. It would be a lot of work to make CodeChecker handle both `str` and `byte` everywhere. Using the `universal_newlines` argument, we can force Python3 to return text rather then bytes.

Reply:
Currently, in Python 2, we also get text with the default encoding (and not bytes) from subprocess calls. This added feature will just preserve the current semantics under Python 3. If we want to change the semantics, I think it should be done in a separate pull requests. But nowadays, I think almost all systems with reasonable settings should have unicode as the default encoding.